File: app/src/main/java/com/dumuzeyn/mp3player/NavigationState.java
package com.dumuzeyn.mp3player;

/** Current library destination and transient tab transition state. */
final class NavigationState implements TabTransitionCoordinator.Navigation {
    int tabIndex;
    int preferredTabDirection;
    boolean tabAnimating;
    String search = "";
    boolean fullPlayerOpening;
    int songRenderGeneration;
    boolean renderingTabPreview;

    @Override
    public int selectedTab() {
        return tabIndex;
    }

    @Override
    public void setSelectedTab(int value) {
        tabIndex = Math.max(0, value);
    }

    @Override
    public String searchQuery() {
        return search;
    }

    @Override
    public void setSearchQuery(String value) {
        search = value == null ? "" : value;
    }

    @Override
    public void setPreferredDirection(int value) {
        preferredTabDirection = value;
    }

    @Override
    public void setTransitionRunning(boolean value) {
        tabAnimating = value;
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/NavigationState.kt
package com.dumuzeyn.mp3player

/** Current library destination and transient tab transition state. */
internal class NavigationState : TabTransitionCoordinator.Navigation {
    @JvmField var tabIndex = 0
    @JvmField var preferredTabDirection = 0
    @JvmField var tabAnimating = false
    @JvmField var search = ""
    @JvmField var fullPlayerOpening = false
    @JvmField var songRenderGeneration = 0
    @JvmField var renderingTabPreview = false

    override fun selectedTab(): Int = tabIndex

    override fun setSelectedTab(value: Int) {
        tabIndex = value.coerceAtLeast(0)
    }

    override fun searchQuery(): String = search

    override fun setSearchQuery(value: String?) {
        search = value.orEmpty()
    }

    override fun setPreferredDirection(value: Int) {
        preferredTabDirection = value
    }

    override fun setTransitionRunning(value: Boolean) {
        tabAnimating = value
    }
}
